src/solenoid/server.clj

```diff
@@ -111,14 +113,27 @@
                        value                                 (if (= :text (keyword control-type))
                                                                (u/get-string-value query-string)
                                                                value)
-                       block-id                              (when block-id (u/stringified-key->keyword block-id))]
-                   (when value (swap! c/registry assoc-in [id :value] value))
+                       block-id                              (when block-id (u/stringified-key->keyword block-id))
+                       dependents                            (->> (get-in @c/registry [block-id :state]) u/get-watches)]
+                   (when (not (nil? value)) (swap! c/registry assoc-in [id :value] value))
                    {:status 200
-                    :body   (str (html (if (= (keyword control-type) :edn) (str value) value))
-                                 (when block-id
-                                   (html (components/render-control-block-result
-                                           (get @c/registry block-id)
-                                           true))))}))}
+                    :body   (apply str
+                                   (concat
+                                     [;; render a controller
+                                      (html (case (keyword control-type)
+                                              :point (components/render-point-value {:value value})
+                                              (str value)))
+                                      ;; when rendering a block, they're updated with hx out-of-band true
+                                      (when block-id
+                                        (html (components/render-control-block-result
+                                                (get @c/registry block-id)
+                                                true)))]
+                                     ;; when there are dependent control blocks, send their results too
+                                     (when (seq dependents)
+                                       (for [[_ block-id] dependents]
+                                         (html (components/render-control-block-result
+                                                 (get @c/registry block-id)
+                                                 true))))))}))}
```

Comment:
Here is where the :dependent mechanism is actually used to discover blocks that are dependent on the currently updating block.